clang/lib/Sema/SemaTemplateVariadic.cpp

```diff
@@ -353,7 +353,11 @@ Sema::DiagnoseUnexpandedParameterPacks(SourceLocation Loc,
       }
 
       if (!EnclosingStmtExpr) {
-        LSI->ContainsUnexpandedParameterPack = true;
+        // It is ok to have unexpanded packs in captures, template parameters
+        // and parameters too, but only the body statement does not store this
+        // flag, so we have to propagate it through LamdaScopeInfo.
+        if (LSI->AfterParameterList)
+          LSI->BodyContainsUnexpandedParameterPack = true;
```

Comment:
Yeah, one thing I didn't notice in #86265 is that we could propagate up the unexpanded pack flag for statements through calls to `DiagnoseUnexpandedParameterPacks()`. (IIUC we call this function at the end of every expressions?)

Reply:
I think this just force us to do more work - going over the lambda twice is not ideal

Reply:
Wait I thought it is my approach that goes through the lambda body twice in a transformation. I didn't see additional calls to `DiagnoseUnexpandedParameterPacks` in this patch?

Reply:
Yeah, there aren't any additional recursive traversals. It's only a shallow traversal over the fields of the lambda itself and it's corresponding members.

Reply:
It's shallow but it's also work that we could avoid doing entirely - at least during parsing just by not having this if.